microsoft_outlook_tenant_id/models/microsoft_outlook_mixin.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class MicrosoftOutlookMixin(models.AbstractModel):
+    _inherit = "microsoft.outlook.mixin"
+
+    @property
+    def _OUTLOOK_ENDPOINT(self):
+        tenant_id = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("microsoft_outlook_tenant_id")
+        )
+        outlook_endpoint = "https://login.microsoftonline.com/common/oauth2/v2.0/"
+        if tenant_id:
+            outlook_endpoint = outlook_endpoint.replace("common", tenant_id)
+        return outlook_endpoint
```

Comment:
```suggestion
    def _OUTLOOK_ENDPOINT(self):
        outlook_endpoint = "https://login.microsoftonline.com/{path}/oauth2/v2.0/"
        path =  (
            self.env["ir.config_parameter"]
            .sudo()
            .get_param("microsoft_outlook_single_tenant.tenant_id", "common")
        )
        return outlook_endpoint.format(path=path)
```

System param adapted to module's new name

Reply:
I'm not 100% ok with this change request.
This means that till the module is installed, it will not be possible anymore to connect with a multi tenant.

Reply:
The endpoint defines if it is a single-tenant or a multi-tenant application.

Reply:
So with the requested change, you need to uninstall the application to be able to switch to multi tenant.
And with your original code, you only need to remove a configuration.

IMO the original code is better

Reply:
@simahawk for the parameter I propose to keep that name of the system param since it is the one used in azure => Directory (tentant) ID
![image](https://user-images.githubusercontent.com/67733397/201665149-ad4e6d21-d531-4071-b915-89b6246c4b20.png)

Reply:
@leemannd the result is the same. "common" is always the fallback in case you don't have the param.
And if the module is uninstalled this code won't run.
If you want to address what you say, you need to store the value somewhere else instead of a config param.

@santostelmo `microsoft_outlook_single_tenant.directory_tenant_id` you mean?

Reply:
Yes `microsoft_outlook_single_tenant.directory_tenant_id`

Reply:
@simahawk Next time I promise to read twice the code before commenting. Sorry for the noise